File: problems/1346/solution.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function(arr) {
    const ar = [];
    
    for (let i = 0; i < arr.length; i++) {
        const val = arr[i];
        
        if (ar.indexOf(val * 2) !== -1) {
            return true;
        }
        
        if (val % 2 === 0 && ar.indexOf(val / 2) !== -1) {
            return true;
        }
        
        ar.push(val);
    }
    
    return false;
};

File: problems/1346/solution.go
package main
func checkIfExist(arr []int) bool {
    for i, n := range arr {
        for j, m := range arr {
            if i == j {
                continue
            }
            
            if n*2 == m || n == m*2 {
                return true
            }
            
            // if (n % 2 == 0 && n/2 == m) || (m % 2 == 0 && m/2 == n) {
            //     return true
            // }
        }
    }
    
    return false
}